client/scripts/views/components/stage_editor.ts

```diff
@@ -249,13 +275,17 @@ const StageEditor: m.Component<{
               } catch (err) {
                 console.log('Failed to update stage');
                 throw new Error((err.responseJSON && err.responseJSON.error)
-                  ? err.responseJSON.error
-                  : 'Failed to update stage');
+                  ?  `${err.responseJSON.error}. Make sure one is selected.`
+                  : 'Failed to update stage, make sure one is selected');
               }
 
               // set linked chain entities
               try {
                 await app.threads.setLinkedChainEntities({ threadId: thread.id, entities: vnode.state.chainEntitiesToSet });
+                for (let i = 0; i < vnode.state.snapshotProposalsToSet.length; i++) {
+                  await app.threads.setLinkedSnapshotProposal({ threadId: thread.id, 
```

Comment:
So we're only linking the *last* snapshot proposal in the list? Because we only support linking a single one.